### PR TITLE
Switching to CocoaPods

### PR DESCRIPTION
**Pull request by czechboy0**
Switching from Carthage to CocoaPods, because we'll need also BuildaUtils as a dependency. And with more dependencies, in my opinion, Carthage starts being too much manual work. 

Just FYI, @esttorhe @cojoj.

### CODE REVIEW

**Comment by esttorhe** [comment]
I found out the same thing; as the dependencies grow `CocoaPods` beats `Carthage` on integration.

Thanks for letting us know @czechboy0.

I'm finishing [`RxViewModel`](https://github.com/esttorhe/RxViewModel) and then hopefully I'll jump back to contributing here :smile:

**Comment by czechboy0** [comment]
I'd actually like to introduce ReactiveCocoa to Buildasaur. Managing the UI turned out to be insanely complex and annoying.

Why are you on the Rx side of things?

**Comment by esttorhe** [comment]
Some time ago when the hype for `ReactiveCocoa` was starting I «decided» that it was going to be the «future» and decided to jump head over heels on the Reactive bandwagon.

I read one of Ash Furrow's books that was focused on `ReactiveViewModel` and got totally hooked up to it.

Now I was working on a personal project on `Swift 2.0` and decided to use `Moya` (didn't have support for 2.0) and went to migrate it; then one of its dependencies was `RxSwift` so I went and integrated the `2.0` branch; the last dependency is `ReactiveCocoa` but migrating that dependency its a whole other story.

Long story short I wanted to use RVM but due to the fact that the project seems left to die it kind of inspired me to create a similar approach using `Rx`.

---

Boy; I digress… I just like reactive-thingies :smile: 
My projects have definitely benefit for the better by including reactive-all-the-things to them.

**Comment by czechboy0** [comment]
Right, but there are two major frameworks: Rx and ReactiveCocoa. Which would you recommend? I just know the basics of ReactiveCocoa.

**Comment by esttorhe** [comment]
So far I have only used `ReactiveCocoa` but `ReactiveViewModel` didn't even had a `Swift 1.x` branch and that's why I decided to give it a try to `Rx`.

RC is based off of Rx (Microsoft extensions) and I have found it easier to understand and absorb vs RC.

You can't go wrong with either of those frameworks but I've been exploring Rx lately more; to me it feels more mature (more so because its kind of «shared» between multiple technologies/languages).

**Comment by czechboy0** [comment]
Interesting, thanks! :+1:

**Comment by cojoj** [comment]
Great! I'm in favour of **Cocoapods** over **Carthage**... 😉

On _reactive_ stuff here... I really have no experience but there's some kinde of hype sround `ReactiveCocoa` and many people think that 3.0 will be a total game-changer as if `RC` was especially designed to work the Swifty way.
